### PR TITLE
chore: migrate to the latest version of `go-control-plane` and use `golang/protobuf` instead of `gogo/protobuf`

### PR DESCRIPTION
**Pull request by yskopets**
### Summary

* to be able to use our L4 subset load balancing feature [contributed to Envoy](https://github.com/envoyproxy/envoy/pull/8870), we need to upgrade dependency on `go-control-plane`
* unfortunately, `go-control-plane` has changed in a backwards-incompatible way in `v0.9.0` (usage of `gogo/protobuf` was replaced with `golang/protobuf`)
* that's why we need to refactor our own code
* additionally, for consistency with `go-control-plane`, we are also moving from `gogo/protobuf` to `golang/protobuf`

### Watch out!

* by dropping `gogo/protobuf` we loose a few of its features we were using
  * it's no longer possible to use `(gogoproto.nullable) = false` option to force code generator to generate `non-nillable` fields (we used this in `DataplaneInsight` and `DataplaneOverview`)
  * so, watch out for `nil`s

### CODE REVIEW

File: api/mesh/v1alpha1/dataplane_insight_helpers.go [hunk 1]
```diff
@@ -4,28 +4,45 @@ import (
...
+		Rds:   &DiscoveryServiceStats{},
+	}
+}
+
 func (ds *DataplaneInsight) IsOnline() bool {
-	for _, s := range ds.Subscriptions {
-		if s.ConnectTime != nil && s.DisconnectTime == nil {
-			return true
+	if ds != nil {
+		for _, s := range ds.GetSubscriptions() {
```
Comment (jakubdyszkiewicz) [inline api/mesh/v1alpha1/dataplane_insight_helpers.go]: doesn't `GetSubscriptions()` handle nil?

File: pkg/xds/server/server.go [hunk 2]
```diff
@@ -93,21 +97,28 @@ func (values watches) Cancel() {
...
 
 func createResponse(resp *cache.Response, typeURL string) (*v2.DiscoveryResponse, error) {
 	if resp == nil {
 		return nil, errors.New("missing response")
 	}
-	resources := make([]*types.Any, len(resp.Resources))
+	resources := make([]*any.Any, len(resp.Resources))
 	for i := 0; i < len(resp.Resources); i++ {
-		data, err := proto.Marshal(resp.Resources[i])
+		// Envoy relies on serialized protobuf bytes for detecting changes to the resources.
```
Comment (jakubdyszkiewicz) [inline pkg/xds/server/server.go]: On serialized bytes? Not on the version of the resource?

File: pkg/xds/server/server.go [hunk 3]
```diff
@@ -246,6 +270,13 @@ func (s *server) process(stream stream, reqCh <-chan *v2.DiscoveryRequest, defau
 				return status.Errorf(codes.Unavailable, "empty request")
 			}
 
+			// node field in discovery request is delta-compressed
```
Comment (jakubdyszkiewicz) [inline pkg/xds/server/server.go]: What do you mean by delta-compressed?

Comment (yskopets) [reply pkg/xds/server/server.go]: copied "as is" from https://github.com/envoyproxy/go-control-plane/blob/34c8be4/pkg/server/server.go#L282

File: pkg/xds/server/server.go (hunk 2 above)
Comment (yskopets) [reply pkg/xds/server/server.go]: copied "as is" from https://github.com/envoyproxy/go-control-plane/blob/34c8be4/pkg/server/server.go#L144

File: api/mesh/v1alpha1/dataplane_insight_helpers.go (hunk 1 above)
Comment (yskopets) [reply api/mesh/v1alpha1/dataplane_insight_helpers.go]: fixed

**Pull request merged**